dell/csm-operator
# Merge in 1.4.2 fixes - verified

Note: In draft until testing is complete 
* Add permissions for SC removal. (#431)

* Bump_up_csm-operator_version_to_1.4.1 (#434)

* Patch version support (#433)

* Adding support of v2.9.1 for Pmax, Pscale, PFlex and unity

* Updating module config version for Pmax, Pscale, PFlex and unity

* Update storage_csm_powerflex.yaml

* version changed to 2.9.1

* version changed to 2.9.1

* Enabling github actions for all branch

* updated version to 2.9.1 for unity, powerflex, powerscale and powermax

* Adding the support of PStore v2.9.1

* Adding the support of PFlex v2.9.1

* Version bumpup replication 1.7.1

* Bump up version for authorization, resiliency

* update csi-metadata-retriever version to 1.6.1

* update reverseproxy version to 2.8.1

---------






* Change namespace name to match documentation. (#437)

* go-yaml version update (#438)

* go-yaml version update

* update go-yaml package to latest

* remove extra verbs

* remove duplicate verbs and add correct test files for resiliency UT

---------



* [bug-1120]: Update Authorization sidecar to use insecure flag (#442)

* fix auth cert

* remove constant

* remove duplicate verbs

* reverting #426 changes (#444)

* Update final manifests for posting of Operator 1.4.1 (#445)

* KRV-20573/csm status (#446)

* add module status

* Adding CSM Module status

* Addressing the comments for PR

* Making code reusable

* Fix formatting issue

* Resolving gosec issue

* Final manifest update and apex client (#447)

* Add csm watch permissions. (#448)

* Merged CSM Module Status

* Fix App-Mob node error (#457)



* Bump up CSM Operator version to 1.4.2 (#458)

* update csm operator version to 1.4.2

* bump-up-op-version-1.4.2

* updated version in operator.yaml

* adding missing arg to fix e2e

* update authorization e2e (#459)

* skip cert validation in e2e (#460)

* Fixing FSGroupPolicy issue (#463)

* Fixing FSGroupPolicy issue

* Add unit test

---------



* Fix for controller race, daemon status, and obs namespace  (#464)

* test controller state calc

* add debug logs

* check for driver

* add daemonnset update

* adds ns fix for when pod ns dis not csm ns

* add ns check for daemon and deployment as well

* fix typo

* fix typo

* explicitly return nil

* add missing replace statements

---------



* Auth proxy status fix (#465)

* actions

* working on auth proxy status check

* finished status check

* fix 1

* fix or

* code compiles

* keep auth proxy out of daemonset check

* trying to figure out deployment name

* formatting

* fix auth name in if condition

* no daemonset

* fix

* added log statements

* added another log statement

* fix log statement

* logs logs logs

* compile

* more logs

* add logs and try updating status from handlesuccess

* remove UpdateStatus call for now

* adding csm label to auth

* remove extraneous string replacement

* removing commented code

* Update actions.yml

* fix merge issue

* fix fmt

* fix pscale + obs issue  (#467)

* fix some issues

* fiix label issue, add debug logs

* add more debug

* more  debugg lines

* fix pscale issue

* remmove debug logs

* gofmt

* set available to 0 en deployment is not checked

* fix if check

---------



* update final manifests (#468)

* fix vetting issues (#477)



---------

# Description
Merges in the fixes made for 1.4.2 into main branch 

# GitHub Issues
List the GitHub issues impacted by this PR:

| GitHub Issue # |
| -------------- |
| https://github.com/dell/csm/issues/1146 |

# Checklist:


# How Has This Been Tested?
Please describe the tests that you ran to verify your changes. Please also list any relevant details for your test configuration

- [x] E2E tests ran
- [x] Unit tests ran
